packages/block-library/src/image/edit.js

```diff
@@ -164,12 +165,19 @@ export function ImageEdit( {
 				url
 			)
 		);
+
 		// If the image block was not replaced with an embed,
+		// and it's not an external image,
+		// and it's been deleted from the database,
 		// clear the attributes and trigger the placeholder.
-		if ( ! isReplaced ) {
-			setAttributes( {
-				url: undefined,
-				id: undefined,
+		if ( id && ! isReplaced && ! isExternalImage( id, url ) ) {
+			isMediaFileDeleted( id ).then( ( isFileDeleted ) => {
```

Comment:
I'm wondering if this will solve the reported issue in https://github.com/WordPress/gutenberg/issues/41161 at all... if the image triggers the error handler, it could be due to a network error in which case the fetch request would fail. 

🤔

Reply:
Is this a situation we could mock within some unit tests for the image block?